e2e/tests/components/button-spec.ts

```diff
@@ -0,0 +1,79 @@
+import { test, expect, devices } from '@playwright/test';
+
+test.describe('Button', () => {
+
+  let componentUrls: string[] = [];
+
+  const workingDirectory = process.cwd();
+  // split the last part of the workingDirectory path and add the storybook path
+  const localStorybookPath = `file://${workingDirectory.split('/').slice(0, -1).join('/')}/packages/react/storybook-static/index.html?path=/story/`;
+  const componentPath = `${localStorybookPath}components-button`;
+
+  [devices['Desktop Chrome'], devices['iPhone SE']].forEach(device => {
+    test(`button ${device.isMobile ? 'mobile' : 'desktop'}`, async ({ page }) => {
+      page.setViewportSize(device.viewport);
+      await page.goto(componentPath);
+
+      if (componentUrls.length == 0) {
+        // get components with data-parent-id="components-button"    
+        await expect(page.locator("#components-button")).toBeVisible();
+        const components = await page.locator('[data-parent-id="components-button"]').all();
+        // loop components and get each's url
+        for (const component of components) {
+          await component.getAttribute('href').then((href) => {
+            // don't add anything containing 'playground' to the list
+            if (href && !href.includes('playground')) {
+              componentUrls.push(href as string);
+            }
+          });
+        }
+      }
+    });
+
+    test(`subTest-${device.isMobile ? 'mobile' : 'desktop'}`, async ({ page }) => {
+      page.setViewportSize(device.viewport);
+      for (const componentUrl of componentUrls) {
+        await page.goto(`file://${componentUrl}`);
+        const iFrame = await page.frameLocator('iframe[title="storybook-preview-iframe"]');
```

Comment:
If the url is opened without the iframe, the tests should be clearer/faster?

iframe.html?args=&id=<testid>&viewMode=story

Reply:
Checking....

Reply:
Well, it's quite the same to formulae the url from the story button link url than to just locate the iframe 🤷

Reply:
Done!